src/blockstack_cli.rs

```diff
@@ -0,0 +1,431 @@
+#![allow(unused_imports)]
+#![allow(unused_assignments)]
+#![allow(unused_variables)]
+#![allow(dead_code)]
+#![allow(non_camel_case_types)]
+#![allow(non_snake_case)]
+#![allow(non_upper_case_globals)]
+
+extern crate blockstack_lib;
+
+use std::{io, fs, env};
+use std::convert::TryFrom;
+use std::io::Read;
+use blockstack_lib::util::{log, strings::StacksString, hash::hex_bytes, hash::to_hex};
+use blockstack_lib::vm;
+use blockstack_lib::vm::{
+    Value, ClarityName, ContractName, errors::RuntimeErrorType, errors::Error as ClarityError };
+use blockstack_lib::chainstate::stacks::{
+    AddressHashMode, C32_ADDRESS_VERSION_MAINNET_SINGLESIG, C32_ADDRESS_VERSION_TESTNET_SINGLESIG,
+    StacksPrivateKey, TransactionSpendingCondition, TransactionAuth, TransactionVersion,
+    StacksPublicKey, TransactionPayload, StacksTransactionSigner,
+    StacksTransaction, TransactionSmartContract, TransactionContractCall, StacksAddress };
+use blockstack_lib::burnchains::Address;
+use blockstack_lib::net::{Error as NetError, StacksMessageCodec};
+
+const USAGE: &str = "blockstack-cli (options) [method] [args...]
+
+This CLI allows you to generate simple signed transactions for blockstack-core
+to process.
+
+This CLI has two methods:
+
+  publish          used to generate and sign a contract publish transaction
+  contract-call    used to generate and sign a contract-call transaction
+  generate-sk      used to generate a secret key for transaction signing
+
+For usage information on those methods, call `blockstack-cli [method] -h`
+
+`blockstack-cli` accepts flag options as well:
+
+   --testnet       instruct the transaction generator to use a testnet version byte instead of MAINNET (default)
+
+";
+
+const PUBLISH_USAGE: &str = "blockstack-cli (options) publish [publisher-secret-key-hex] [fee-rate] [nonce] [contract-name] [file-name.clar]
+
+The publish command generates and signs a contract publish transaction. If successful,
+this command outputs the hex string encoding of the transaction to stdout, and exits with
+code 0";
+
+const CALL_USAGE: &str = "blockstack-cli (options) contract-call [origin-secret-key-hex] [fee-rate] [nonce] [contract-publisher-address] [contract-name] [function-name] [args...]
+
+The contract-call command generates and signs a contract-call transaction. If successful,
+this command outputs the hex string encoding of the transaction to stdout, and exits with
+code 0
+
+Arguments are supplied in one of two ways: through script evaluation or via hex encoding
+of the value serialization format. The method for supplying arguments is chosen by
+prefacing each argument with a flag:
+
+  -e  indicates the argument should be _evaluated_
+  -x  indicates the argument is supplied as a hexstring of the value serialization
+
+e.g.,
+
+   blockstack-cli contract-call $secret_key SPJT598WY1RJN792HRKRHRQYFB7RJ5ZCG6J6GEZ4 10 0 foo-contract \\
```

Comment:
This example doesn't match the usage.  What does the address `SPJT598WY1RJN792HRKRHRQYFB7RJ5ZCG6J6GEZ4` correspond to?